addons/web_editor/controllers/main.py

```diff
@@ -317,7 +323,17 @@ def _attachment_create(self, name='', data=False, url=False, res_id=False, res_m
         else:
             raise UserError(_("You need to specify either data or url to create an attachment."))
 
-        attachment = request.env['ir.attachment'].create(attachment_data)
+        env = request.env['ir.attachment']
+        if request.env.user.has_group('base.group_portal'):
+            request.env[res_model].check_access_rights('write')
+            if res_id:
+                request.env[res_model].browse(res_id).check_access_rule('write')
+            env = env.sudo()
+        attachment = env.create(attachment_data)
```

Comment:
Don't overwrite the `env` variable with `env.sudo()`. Just write `env.sudo()` instead of `env` when you need to. I'd rather have this:
```python
if ...:
    ....
    attachment = env.sudo().create(attachment_data)
else:
    attachment = env.create(attachment_data)
```
So that there is no risk of propagating a `sudo` environment by mistake in the future.

Reply:
@dmo-odoo It should be correct now